### PR TITLE
Do not install Rust if the executed task does not need it

### LINKED ISSUE #2474: Optimize makefile
# Problem

The "problem" with our makefile is that we are using global variables.
Global variables in make are evaluated during makefile's "parsing" stage, not during execution stage.
And hence, the toolchain is always installed for any make target.

# Goal

Only install cargo/toolchain for make targets where it is actually needed

<details>
<summary>Notes: </summary>

To compare, check the `Run e2e test` action

No toolchain installation after changes in this branch: 
https://github.com/comit-network/comit-rs/pull/2469/checks?check_run_id=593882267

```
Run export PATH=$HOME/.cargo/bin:$HOME/.local/bin:$HOME/go/bin:$PATH
yarn install v1.22.4
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 0.55s.
yarn run v1.22.4
$ tsc && jest --forceExit
```

Toolchain being installed before changes in another branch: 
https://github.com/comit-network/comit-rs/pull/2459/checks?check_run_id=590941543

```
Run export PATH=$HOME/.cargo/bin:$HOME/.local/bin:$HOME/go/bin:$PATH
error: toolchain '1.42.0-x86_64-unknown-linux-gnu' is not installed
error: toolchain 'nightly-2020-01-15-x86_64-unknown-linux-gnu' is not installed
info: syncing channel updates for '1.42.0-x86_64-unknown-linux-gnu'
info: latest update on 2020-03-12, rust version 1.42.0 (b8cedc004 2020-03-09)
info: downloading component 'cargo'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: installing component 'cargo'
info: installing component 'rust-std'
info: installing component 'rustc'
(cd ./api_tests; yarn install; yarn ci)
```

</details>

### PR DESCRIPTION
**Pull request by D4nte**
Edit: Does not c.l.o.s.e #2474 due to debate present in the issue but does the foundation work.

Note: We still have few errors but I don't think it matters:
```
error: toolchain '1.42.0-x86_64-apple-darwin' is not installed
error: toolchain 'nightly-2020-01-15-x86_64-apple-darwin' is not installed
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,10 +14,18 @@ GIT_HOOKS = $(wildcard $(GIT_HOOKS_PATH)/*)
 INSTALLED_TOOLCHAINS = $(shell $(RUSTUP) toolchain list)
 INSTALLED_COMPONENTS = $(shell $(RUSTUP) component list --installed --toolchain $(TOOLCHAIN))
 INSTALLED_NIGHTLY_COMPONENTS = $(shell $(RUSTUP) component list --installed --toolchain $(NIGHTLY_TOOLCHAIN))
-AVAILABLE_CARGO_COMMANDS = $(shell $(CARGO) --list)
```
Comment (bonomat) [inline Makefile:16]: rust was only installed because of this command? 😅

**Comment by bonomat** [comment]
I don't understand how this solved the problem but maybe my understanding was wrong. Can you shed some light on how this is now fixed? And if, feel free to re-add the CI target into the makefile again.

File: Makefile (hunk 1 above)
Comment (D4nte) [reply Makefile:16]: Yes :) `$(CARGO)` includes the installation of a toolchain.

**Comment by D4nte** [comment]
> I don't understand how this solved the problem but maybe my understanding was wrong. Can you shed some light on how this is now fixed?

`$(CARGO)` is `$(RUSTUP) run --install $(TOOLCHAIN) cargo --color always`
As you can see, `--install $(TOOLCHAIN)` is passed to the `rustup` command, which installs the toolchain if not present.

> Feel free to re-add the CI target into the makefile again.

Yeah, I can see in the original issue it's a controversial debate so not sure I want to risk a debate here. Let's get this PR in and we can debate further in the issue.

**Comment by da-kami** [comment]
Note: I am fine with these changes, hence approved :)

Thomas' comment https://github.com/comit-network/comit-rs/issues/2474#issuecomment-615119906 in the issue argues against merging this though, as far as I understand it.

**Pull request merged**